src/herder/HerderTests.cpp

```diff
@@ -1123,7 +1123,9 @@ TEST_CASE("In quorum filtering", "[herder]")
     node3Config.QUORUM_SET.validators.emplace_back(extraK[2].getPublicKey());
     node3Config.QUORUM_SET.validators.emplace_back(extraK[3].getPublicKey());
 
-    sim->addNode(node3Config.NODE_SEED, node3Config.QUORUM_SET, &node3Config);
+    auto node3 = sim->addNode(node3Config.NODE_SEED, node3Config.QUORUM_SET,
+                              &node3Config);
+    node3->start();
```

Comment:
mmm, looks like we're missing the call to `start` in many other places?
I imagine that after calling `addNode` we should either call `startAllNodes` or call `start` manually?

Maybe add an assert in the simulation class to catch non started applications (not sure how to check for that though)? I think tests kinda work here by accident.

Reply:
From what I see, other test cases call either `start` or `startAllNodes`, except this one, "In quorum filtering" (Not counting the nodes generated in `Topologies.cpp`, since it doesn't really make sense to start them there)

Also, just noticed that some nodes added in earlier in this test are also not started, will update. Regarding the asserts, maybe we should throw in `Simulation::crankNode` (which is used by other cranking functions) if the app is not started.

Reply:
`crankNode` sounds like a perfect place to check for that yes, should be easy to find the bad ones with the assert :)

Reply:
ok, updated. We were not even using initial application state, so I updated `Application::getState`